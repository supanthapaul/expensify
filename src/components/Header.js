import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
    <div>
        <h1>Expensify</h1>
        <ul>
            <li><NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink></li>
            <li><NavLink to="/create" activeClassName="is-active">Create Expense</NavLink></li>
            <li><NavLink to="/help" activeClassName="is-active">Help</NavLink></li>
        </ul>
    </div>
);

export default Header;