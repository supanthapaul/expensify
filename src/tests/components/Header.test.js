import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('Should render Header correctly', () => {
   //snapshot testing
   // https://facebook.github.io/jest/docs/en/expect.html#tomatchsnapshotpropertymatchers-snapshotname

   // Enzyme docs - http://airbnb.io/enzyme/docs/api/shallow.html
   const wrapper = shallow(<Header />);
   expect(wrapper).toMatchSnapshot();
});


