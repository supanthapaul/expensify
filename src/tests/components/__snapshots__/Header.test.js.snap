// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div>
  <h1>
    Header
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/create"
      >
        Create Expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/help"
      >
        Help
      </NavLink>
    </li>
  </ul>
</div>
`;
